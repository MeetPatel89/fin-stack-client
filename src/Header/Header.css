@import url('https://fonts.googleapis.com/css2?family=Fraunces&family=Noto+Sans+JP&display=swap" rel="stylesheet');

.logo {
  background-color: #f7f0f5;
  padding: 10px;
}

h1 a {
  color: #6a2e35;
  font-family: 'Fraunces', serif;
  font-size: 35px;
}

h1 {
  display: inline-block;
}

header {
  display: flex;
  justify-content: space-between;
}

.user {
    display: flex;
    align-items: center;
    font-size: 20px;
    color: #6a2e35;
}