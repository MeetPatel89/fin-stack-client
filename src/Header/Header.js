import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './Header.css';

export default function Header(props) {
  return (
    <>
      <header className='logo'>
        <h1 className='main-heading'>
          <Link to='/'>Fin Stack</Link>
        </h1>
        <span className='user'>{props.user && `Welcome ${props.user}!`}</span>
      </header>
    </>
  );
}

Header.propTypes = {
  user: PropTypes.string,
};

Header.defaultProps = {
  user: 'RandomUser89',
};
