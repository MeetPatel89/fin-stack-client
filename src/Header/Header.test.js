// make React available
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

// make the ReactDOM available necessary for rendering the component
import ReactDOM from 'react-dom';

// make the Header component available
import Header from './Header';

describe('Header Component', () => {
  // this is the test case
  it('renders component without crashing', () => {
    // first create a DOM element to render the component into
    const div = document.createElement('div');

    // render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
      div
    );

    // clean up code
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const treeToRender = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      )
      .toJSON();
    expect(treeToRender).toMatchSnapshot();
  });

  it('renders the UI as expected with user prop', () => {
    const treeToRender = renderer
      .create(
        <BrowserRouter>
          <Header user='Random User' />
        </BrowserRouter>
      )
      .toJSON();
    expect(treeToRender).toMatchSnapshot();
  });
});
