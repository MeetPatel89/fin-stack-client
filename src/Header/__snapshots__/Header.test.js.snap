// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders the UI as expected 1`] = `
<header
  className="logo"
>
  <h1
    className="main-heading"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Fin Stack
    </a>
  </h1>
  <span
    className="user"
  >
    
  </span>
</header>
`;

exports[`Header Component renders the UI as expected with user prop 1`] = `
<header
  className="logo"
>
  <h1
    className="main-heading"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Fin Stack
    </a>
  </h1>
  <span
    className="user"
  >
    Welcome Random User!
  </span>
</header>
`;
